chat-app/src/ChatRoom.js

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react';
+import { MessageList } from './MessageList';
+import { MessageForm } from './MessageForm';
+
+export const ChatRoom = ({ roomId, roomName, username, socket }) => {
+  const [messages, setMessages] = useState([]);
+  const [newMessage, setNewMessage] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const loadMessages = async () => {
+    try {
+      const response = await fetch(`http://localhost:3005/rooms/${roomId}/messages`);
+      if (!response.ok) {
+        throw new Error('Ошибка при загрузке сообщений');
+      }
+      const data = await response.json();
+      setMessages(data);
+    } catch (error) {
+      console.error('Loading error:', error);
+      setErrorMessage('Message download failed');
```

Comment:
Consider providing more detailed error messages to the user when message loading fails. This can help users understand the issue better.